src/app/services/actions/permissions-action-creator.ts

```diff
@@ -0,0 +1,101 @@
+import { IAction } from '../../../types/action';
+import { IRequestOptions } from '../../../types/request';
+
+import { GET_CONSENT_ERROR, SCOPES_FETCH_ERROR, SCOPES_FETCH_SUCCESS } from '../redux-constants';
+import { authenticatedRequest, parseResponse, queryResponse } from './query-action-creator-util';
+
+export function fetchScopesSuccess(response: object): IAction {
+  return {
+    type: SCOPES_FETCH_SUCCESS,
+    response,
+  };
+}
+
+export function fetchScopesError(response: object): IAction {
+  return {
+    type: SCOPES_FETCH_ERROR,
+    response,
+  };
+}
+
+export function getConsentError(response: object): IAction {
+  return {
+    type: GET_CONSENT_ERROR,
+    response,
+  };
+}
+
+export function fetchScopes(): Function {
+  return async (dispatch: Function, getState: Function) => {
+    const query = getState().sampleQuery;
+    const { sampleUrl, selectedVerb } = query;
+    const urlObject: URL = new URL(sampleUrl);
+    // remove the prefix i.e. beta or v1.0 and any possible extra '/' character at the end
+    const requestUrl = urlObject.pathname.substr(5).replace(/\/$/, '');
+    const permissionsUrl = 'https://graphexplorerapi.azurewebsites.net/api/GraphExplorerPermissions?requesturl=' +
+      requestUrl + '&method=' + selectedVerb;
+
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    const options: IRequestOptions = { headers };
+
+    return fetch(permissionsUrl, options)
+      .then(res => res.json())
+      .then(res => {
+        if (res.error) {
+          throw (res.error);
+        }
+        dispatch(fetchScopesSuccess(res));
+      })
+      .catch(() => {
+        const response = {
+          /* Return 'Forbidden' regardless of error, as this was a
```

Comment:
It might be confusing to the customer to display an error we receive from the Permissions API service, e.g. 404, or 500. If we fail to get scopes back it essentially means our request to the resource to Graph is/will be forbidden. I think the permissions tab will be the best place to provide more details on whether a particular sample request url has no scopes, or that the url is not defined in the list of permissions.